### PR TITLE
Refactor Users#index to paginate and filter on the backend. Closes #266

### LINKED ISSUE #266: Implement pagination & search for users list
The users list (shown below) has two problems that we want to tackle: first, it does filtering on the client side, resulting in poor performance on mobile devices for large lists. Secondly, it is not consistent with the rest of the app, where the filtering happens on the backend.

users

Also, we would like to get rid of Angular to cut off some complexity that we think is not needed. It can greatly simplify the app and make it easier for others to contribute to the project.

## Requirements

- [ ] Paginate the response in the backend, so that not all records are returned in one go. This is already done in offers. See image below:

offers

- [ ] Change the search field to query Elasticsearch, rather than filtering on the client side. Again this is already done in `offers/dashboard`.

- [ ] Remove all the now unused Angular machinery. We might not be able to completely remove Angular as it might be used in some other corner of the app.

This clearly fits in two isolated PRs that would go together in the next release.

### PR DESCRIPTION
**Pull request by knoopx**
* Results are now paginated using kaminari on the server side
* Filtering and sorting is performed now on the server side using ransack, which was already part of the bundle (activeadmin depends on it) and because, in my opinion, elasticsearch was kind of overkill for this specific case
* Removed angular completely (couldn't find any other usage within the app)

### CODE REVIEW

**Comment by sauloperez** [comment]
Wellcome to TimeOverflow @knoopx ! keep it coming, it looks pretty good so far 🎉 !

**Comment by sauloperez** [comment]
I thought the login had some piece angular too

**Comment by sseerrggii** [comment]
Thanks!! I see 3 things after testing:

* Top Nav dropdown
* Order users by balance
* Maybe in search box we need a bottom with :mag_right: 

peek 20-10-2017 16-11

**Comment by markets** [comment]
Hi guys :wave:, about:

> The top nav dropdown doesn't open

I think this can be related: https://github.com/coopdevs/timeoverflow/commit/278f0d4dca846f1385e5fa20399f581bd7d140a2

Thanks a lot @knoopx for this PR!

**Comment by sseerrggii** [comment]
:clap::clap::clap:  @knoopx 

Could it be possible to sort by balance? administrators use it... It just did not work well

**Comment by sseerrggii** [comment]
> Could it be possible to sort by balance? administrators use it... It just did not work well

@sauloperez :pray:

File: app/helpers/users_helper.rb [hunk 1]
```diff
@@ -1,34 +1,14 @@
...
-        edit_link: edit_user_path(user),
-        cancel_link: cancel_member_path(membership),
-        toggle_manager_link: toggle_manager_member_path(membership),
-        manager: !!membership.manager,
-        toggle_active_link: toggle_active_member_path(membership),
-        active: membership.active?,
-        valid_email: user.has_valid_email?
-      }
-    end.to_json.html_safe
-  end
```
Comment (sauloperez) [inline app/helpers/users_helper.rb]: We need to make sure we also get rid of the private methods this might use.

**Comment by sauloperez** [comment]
I'm taking this over. Sorry @knoopx, we need to finish it up to move on to new things.

**Comment by sauloperez** [comment]
Done! 

sort

**Comment by enricostano** [comment]
@sseerrggii @sauloperez do the changes in locales files will need some processing in localeapp?

**Comment by sauloperez** [comment]
Given the nature of the change, I guess not, but we'll take a look.

**Comment by sseerrggii** [comment]
fireshot capture 8 - banco de tiempo local - http___local timeoverflow org_3000_members
A scroll tab has appeared, in chrome

**Comment by sauloperez** [comment]
:see_no_evil: I'll take a look.

**Comment by sauloperez** [comment]
Done @sseerrggii. Although was a preexistent issue I managed to fix easily. Can you take another look?

**Comment by sauloperez** [comment]
Tested and working. Moving on!

**Pull request merged**